addons/l10n_tr_nilvera_einvoice/models/account_move.py

```diff
@@ -0,0 +1,253 @@
+                    continue
+                move = self._l10n_tr_nilvera_get_invoice_from_uuid(client, document_uuid)
+                self._l10n_tr_nilvera_add_pdf_to_invoice(client, move, document_uuid)
+
+    def _l10n_tr_nilvera_get_invoice_from_uuid(self, client, document_uuid):
+        response = client.request(
+            "GET",
+            f"/einvoice/Purchase/{document_uuid}/xml",
+        )
+
+        journal = self.env.company.l10n_tr_nilvera_purchase_journal_id
+        if not journal:
+            journal = self.env['account.journal'].search([
+                *self.env['account.journal']._check_company_domain(self.env.company),
+                ('type', '=', 'purchase')
+            ], limit=1)
+
+        attachment_vals = {
+            'name': 'attachment.xml',
+            'raw': response,
+            'type': 'binary',
+            'mimetype': 'application/xml',
+        }
+
+        try:
+            attachment = self.env['ir.attachment'].create(attachment_vals)
+            move = journal.with_context(
+                default_move_type='in_invoice',
+                default_l10n_tr_nilvera_uuid=uuid,
+            )._create_document_from_attachment(attachment.id)
+
+            # If move creation was successful, update the attachment name with the bill reference.
+            if move.ref:
+                attachment.name = f'{move.ref}.xml'
+
+            move._message_log(body=_('Nilvera document has been received successfully'))
+        except Exception:   # noqa: BLE001
+            # If the invoice creation fails, create an empty invoice with the attachment. The PDF will be
+            # added in a later step as well.
+            move = self.env['account.move'].create({
+                'move_type': 'in_invoice',
+                'company_id': self.env.company.id,
+                'l10n_tr_nilvera_uuid': uuid,
+            })
+            attachment_vals.update({
+                'res_model': 'account.move',
+                'res_id': move.id,
+            })
+            self.env['ir.attachment'].create(attachment_vals)
+
+        return move
+
+    def _l10n_tr_nilvera_add_pdf_to_invoice(self, client, invoice, document_uuid):
+        response = client.request(
+            "GET",
+            f"/einvoice/Purchase/{document_uuid}/pdf",
+        )
+
+        filename = f'{invoice.ref}.pdf' if invoice.ref else 'Nilvera PDF.pdf'
+
+        attachment = self.env['ir.attachment'].create({
+            'name': filename,
+            'res_id': invoice.id,
+            'res_model': 'account.move',
+            'datas': response,
+            'type': 'binary',
+            'mimetype': 'application/pdf',
+        })
+
+        if invoice.message_main_attachment_id and \
+                invoice.message_main_attachment_id.name.endswith('.xml') and \
+                'pdf' not in invoice.message_main_attachment_id.mimetype:
```

Comment:
i would prefer either oneline or use a variable that having this on three line but i will not block for that :smile: